sdk/storage/azure-storage-blob/V8_V12.md

```diff
@@ -0,0 +1,301 @@
+    SharedAccessAccountPermissions.ADD);
+
+IPRange ipR = new IPRange("0.0.0.0", "255.255.255.255");
+
+Calendar calendar = Calendar.getInstance();
+Date startDate = calendar.getTime();
+calendar.add(Calendar.DAY_OF_YEAR, 1);
+Date expiryDate = calendar.getTime();
+
+SharedAccessHeaders headers = new SharedAccessBlobHeaders();
+headers.setCacheControl("cache");
+headers.setContentDisposition("dispoistion");
+headers.setContentEncoding("encoding");
+headers.setContentLanguage("language");
+headers.setContentType("type");
+
+SharedAccessAccountPolicy policy = new SharedAccessAccountPolicy();
+policy.setPermissions(permissions);
+policy.setSharedAccessStartTime(startDate);
+policy.setSharedAccessExpiryTime(expiryDate);
+policy.setServices(EnumSet.of(SharedAccessAccountService.BLOB));
+
+// build the token
+String sasToken = SharedAccessSignatureHelper.generateSharedAccessSignatureHashForBlobAndFile(
+    policy, headers, identifier, SharedAccessAccountResourceType.OBJECT.name(), ipR, SharedAccessProtocols.HTTPS_ONLY, blobClient, SharedAccessAccountService.BLOB.name(), null);
+```
+
+In V12, SAS tokens are generated off the client to the resource you wish to generate the SAS for.
+The resource path handled automatically because of this, and the `SharedKeyCredential` authenticating the client is also used automatically to sign the SAS. See [building a client](#buildclient) for how to get an authenticated client.
+```java
+// specify token properties
+String identifier = "identifier";
+BlobSASPermission permissions = new BlobSASPermission()
+    .withRead(true)
+    .withCreate(true)
+    .withDelete(true)
+    .withWrite(true); // We can also choose BlobContainerSasPermission, BlobServiceSasQueryParameters based on the object level we want to grant access.
+OffsetDateTime startTime = OffsetDateTime.now().minusDays(1);
+OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+SasIpRange ipRange = new SasIpRange()
+    .setIpMax("0.0.0.0")
+    .setIpMax("255.255.255.255");
+SasProtocol sasProtocol = SasProtocol.HTTPS_HTTP;
+String cacheControl = "cache";
+String contentDisposition = "disposition";
+String contentEncoding = "encoding";
+String contentLanguage = "language";
+String contentType = "type";
+String version = BlobServiceVersion.V2019_02_02.getVersion();
+
+// build the token
+BlobServiceSasSignatureValues sasSignatureValues = new BlobServiceSasSignatureValues(version, sasProtocol, startTime, expiryTime, permissions.toString(), ipRange, identifier, cacheControl, contentDisposition, contentEncoding, contentLanguage, contentType);
+BlobServiceSasQueryParameters sasQueryParameters = sasSignatureValues.generateSasQueryParameters(new StorageSharedKeyCredential("<account-name>", "<account-key"));
+```
+
+## Build Client
+In V8, core classes were built from `CloudStorageAccount` which authenticated with connection string.
+```java
+CloudStorageAccount storageAccount = CloudStorageAccount.parse("<connection-string>");
```

Comment:
It would be good to show CloudBlobClient here instead so it's more of a direct comparison

Reply:
It would also be good I think to show direct construction of CloudBlobClient since we are using the CloudBlobClient builder instead of going through the BlobServiceClient

Reply:
Could you show me how to get CloudBlockBlob through the CloudBlobClient?